datasets/makhzan/makhzan.py

```diff
@@ -0,0 +1,130 @@
+# coding=utf-8
+# Copyright 2020 The HuggingFace Datasets Authors and the current dataset script contributor.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""An Urdu text corpus for machine learning, natural language processing and linguistic analysis."""
+
+from __future__ import absolute_import, division, print_function
+
+import logging
+import xml.etree.ElementTree as ET
+
+import datasets
+
+
+# Find for instance the citation on arxiv or on the dataset repo/website
+_CITATION = """\
+@InProceedings{huggingface:dataset,
+title = {A great new dataset},
+authors={huggingface, Inc.
+},
+year={2020}
+}
+"""
+
+# You can copy an official description
+_DESCRIPTION = """\
+An Urdu text corpus for machine learning, natural language processing and linguistic analysis.
+"""
+
+_HOMEPAGE = "https://matnsaz.net/en/makhzan"
+
+_LICENSE = "All files in the /text directory are covered under standard copyright. Each piece of text has been included in this repository with explicity permission of respective copyright holders, who are identified in the <meta> tag for each file. You are free to use this text for analysis, research and development, but you are not allowed to redistribute or republish this text. Some cases where a less restrictive license could apply to files in the /text directory are presented below. In some cases copyright free text has been digitally reproduced through the hard work of our collaborators. In such cases we have credited the appropriate people where possible in a notes field in the file's metadata, and we strongly encourage you to contact them before redistributing this text in any form. Where a separate license is provided along with the text, we have provided corresponding data in the publication field in a file's metadata."
+
+# The HuggingFace dataset library don't host the datasets but only point to the original files
+# This can be an arbitrary nested dict/list of URLs (see below in `_split_generators` method)
+_BASE_URL = "https://raw.githubusercontent.com/zeerakahmed/makhzan/master/text/"
+_URLs = {"train": [_BASE_URL + "{:04d}.xml".format(i) for i in range(1, 5523)]}
```

Comment:
Can you download all the files at once ?
For example you can download the zip file at https://github.com/zeerakahmed/makhzan/archive/master.zip

Then in the _generate_examples method you can iterate over all the xml files by using `glob` for example.

This way in your dummy_data.zip file you would only need to add 1 or 2 xml files instead of 5000, since glob will only find 1 or 2 files.

Reply:
That's a smart idea. I'll update my code to fix that. :)

Reply:
Thanks :) Feel free to ping me when you're ready for a review

Reply:
Thanks Quentin. I’ve made the requested changes. Tests for dummy data passed locally. The CI tests are failing. I’ll check, fix and let you know.